src/compiler/Stage/Desugar.elm

```diff
@@ -146,6 +149,13 @@ desugarLet recurse bindings body =
         (recurse body)
 
 
+desugarList : (Frontend.Expr -> Result DesugarError Canonical.Expr) -> List Frontend.Expr -> Result DesugarError Canonical.Expr
+desugarList recurse list_ =
+    List.map recurse list_
+        |> List.foldr (Result.map2 (::)) (Ok [])
+        |> Result.map Canonical.List
```

Comment:
Again `list_` - can be `list` or `items`

Just to check my understanding - `foldr` is here to not reverse the list accidentally?